### ISSUE TITLE
Request access for Cython

### ISSUE THREAD

**Issue by lysnikolaou** [scientific-python/upload-nightly-action#80]
Hi all, now that there's ongoing work on making Cython work with the free-threaded build of CPython, it'd be nice for the other projects that are trying to do the same (NumPy, SciPy, scikit-learn, etc.) to be able to install Cython nightly.

The extra load should be minimal, since we only plan to upload the non-compiled wheel, which means we'll only upload one wheel per version, so no platform-specific wheels.

Cython issue: https://github.com/cython/cython/issues/6223

**Comment by tupui** [comment]
Hi @lysnikolaou, when you say non compiled wheel, does it mean someone could just install main with `pip`? Or is there something specific to do?

Otherwise, in general I am +1 for Cython.

If your request goes through, you will need to provide us with anaconda handles of the maintainers you want to have in the team over there.

**Comment by matthewfeickert** [comment]
@lysnikolaou I think this sounds good, so assuming no one has objections, please complete the missing items from the checklist to get started.

* [ ] Create an account on [Anaconda Cloud](https://anaconda.org/)
* [ ] Reply here with the link to your account
* [ ] Make a new Anaconda Cloud API token with permissions described in https://scientific-python.org/specs/spec-0004/#process-for-adding-new-projects

![image](https://github.com/scientific-python/upload-nightly-action/assets/5142394/697ce89f-4a51-43d0-a26f-2493f4bdce34)

* [ ] Save the API token as a repo GitHub Actions secret named `ANACONDA_ORG_UPLOAD_TOKEN`
* [ ] Direct me (@matthewfeickert) to a `cython` nightly wheel for me to upload to create the package in the registry
* [ ] Verify that I have added you to the [`cython` team "Group"](https://anaconda.org/scientific-python-nightly-wheels/groups/cython) on https://anaconda.org/scientific-python-nightly-wheels
* [ ] Verify that I have added the package `scientific-python-nightly-wheels/cython` to the [`cython` team "Group"](https://anaconda.org/scientific-python-nightly-wheels/groups/cython)

If you'd also like to use the [`scientific/upload-nightly-action` GitHub Action](https://github.com/scientific-python/upload-nightly-action) please check out [the action repo's README](https://github.com/scientific-python/upload-nightly-action?tab=readme-ov-file#nightly-upload). If you need to collect wheels from different CI jobs, consider creating a new GitHub Actions workflow with logic similar to [`matplotlib`'s](https://github.com/matplotlib/matplotlib/blob/88d64e5f0cc82a050b840e61ac60e1081bb6d799/.github/workflows/nightlies.yml) or [`Awkward`'s](https://github.com/scikit-hep/awkward/pull/3012).

If you have any questions please just ask here!

**Comment by lysnikolaou** [comment]
> when you say non compiled wheel, does it mean someone could just install main with `pip`? Or is there something specific to do?

I'm not sure if I understand the question, but my understanding is this (correct me if I'm wrong):

Cython basically ships two different types of wheels. Firstly, note that Cython, the compiler, is basically a pure-Python package. One set of wheels, where the Python modules of Cython itself get compiled with Cython and therefore the wheels have to be platform and version-specific. Additionally, there's a version where the Python sources are shipped themselves without any sort of compilation, therefore that is a `py3-none-any` wheel. The only difference between the two is basically performance, like @da-woods described [on the Cython issue](https://github.com/cython/cython/issues/6223#issuecomment-2140304534).

> I think this sounds good, so assuming no one has objections, please complete the missing items from the checklist to get started.

Since I'm not a maintainer of Cython, it'd be best if @scoder or @da-woods did this. I'm just helping set this up and I'll open a Cython PR with the needed CI shortly.

**Comment by tupui** [comment]
To clarify, the original intent of this anaconda channel is to provide a place where wheels of difficult projects to build (in our ecosystem) can be hosted. We are in a better situation as our quota was increased, but it still good if we are conscious about that.

TL;DR if someone can just do `pip install git+https://github.com/...` then to me there is little advantage to use this conda channel.

That being said, considering the importance of Cython in our ecosystem, I would personally don't mind 😃

**Comment by da-woods** [comment]
> TL;DR if someone can just do `pip install git+https://github.com/...` then to me there is little advantage to use this conda channel.

or `NO_CYTHON_COMPILE=true pip install git+https://github.com/...` if they don't want to spend time building Cython (just emphasising this because I think it makes sense for people who just want to test the current master).

But yes - now you mention it I'm slightly unclear what a wheel gives that a link to github doesn't.

I'm definitely keen to encourage/help people test with the latest version of Cython so I'll go through the steps above if people do want a wheel.

**Comment by matthewfeickert** [comment]
> I'm definitely keen to encourage/help people test with the latest version of Cython so I'll go through the steps above if people do want a wheel.

That's enough in my mind, so please feel free to proceed.

**Comment by lysnikolaou** [comment]
> now you mention it I'm slightly unclear what a wheel gives that a link to github doesn't

FWIW it gives people the ability to do `pip install -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple cython` in their CI without having to install from master directly (probably a few more breakages than wheels built twice a week, plus people generally avoid it) and also not pin to a specific commit that they'll have to go back and change afterwards. So more people will be encouraged to test early (both Cython and the free-threaded build itself).

**Comment by da-woods** [comment]
Right - I've created an Anaconda account with username da-woods.

**Comment by matthewfeickert** [comment]
@lysnikolaou I've got everything setup on the backend now, so if you can get the API token and then a workflow to upload the nightly wheels you're good to go. :+1:

**Comment by lysnikolaou** [comment]
Thanks @matthewfeickert! @da-woods Will you please add the repository secret so that we can test cython/cython#6229?

**Comment by matthewfeickert** [comment]
@lysnikolaou if you have an Anaconda cloud account I can add you to the team as well, which would be preferred so that package teams on the nightly index don't have just a single maintainer.

**Comment by lysnikolaou** [comment]
@matthewfeickert The second person should probably be @scoder. I'm okay to be added but that decision should lie with the maintainers.

**Comment by matthewfeickert** [comment]
Thanks. 👍 @scoder do you have an Anaconda Cloud account?

**Comment by matthewfeickert** [comment]
PR https://github.com/cython/cython/pull/6229 is in and wheels got uploaded, so closing as everything looks good. :+1:
